File: src/trackers/es/hachede_me.ts
import {WebChecker} from "../../checker/webChecker";
import {CheckerResult} from "../../checker/checkerResult";

export default class Hachede_me extends WebChecker {
    public get url() { return "https://hachede.me/"; }
    public get name() { return "hachede.me"; }
    public get language() { return "es"; }
    public get categories(): Array<String> { return ["movies"]; }
    public get description() { return ""; }
    public get registryUrl() { return "https://hachede.me/?p=signup&pid=16"; }

    async isOpen(): Promise<CheckerResult> {
        let $ = await WebChecker.getQueryUrl(this.registryUrl);
        let exists = $(":contains('Lo sentimos, pero en estos momentos los registros')").length > 0;

        return {
            online: true,
            regClosed: !exists,
            regOpened: false
        };
    }
}

File: checker/trackers/hachedeme.go
package trackers

import (
	"github.com/tokkenno/open-torrent-tracker-bot/checker/check"
	"github.com/tokkenno/open-torrent-tracker-bot/models"
	"strings"
)

type HachedeMe struct {
}

func (tracker *HachedeMe) GetId() string {
	return "hachede_me"
}

func (tracker *HachedeMe) GetName() string {
	return "+HacheDe"
}

func (tracker *HachedeMe) GetDescription() string {
	return ""
}

func (tracker *HachedeMe) GetLanguages() []string {
	return []string{"es"}
}

func (tracker *HachedeMe) GetCategories() []string {
	return []string{"movies"}
}

func (tracker *HachedeMe) GetUrl() string {
	return "https://hachede.me"
}

func (tracker *HachedeMe) GetRegistryUrl() string {
	return "https://hachede.me/?p=signup&pid=16"
}

func (tracker *HachedeMe) Check() *check.Result {
	html, err := check.GetWebCode(tracker.GetRegistryUrl())

	if err != nil {
		return &check.Result{
			IsOnline: false,
		}
	} else {
		var status models.RegistrationStatus

		if strings.Contains(html, "Lo sentimos, pero en estos momentos los registros están cerrados") {
			status = models.RegistrationClose
		} else {
			status = models.RegistrationMaybe
		}

		return &check.Result{
			IsOnline: true,
			Status:   status,
		}
	}
}
